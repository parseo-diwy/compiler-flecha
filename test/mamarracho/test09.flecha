
def test x y =
  unsafePrintChar 'T';
  unsafePrintChar 'E';
  unsafePrintChar 'S';
  unsafePrintChar 'T';
  unsafePrintChar ' ';
  unsafePrintInt x;
  unsafePrintChar '\n';
  unsafePrintChar '\t';
  unsafePrintInt y;
  unsafePrintChar '\n'

def true x y = x

def false x y = y

def cond x y z = x y z

def main =
 test 1 ((\ x y -> x) 42 43);
 test 2 ((\ x y -> y) 42 43);
 test 3 (true 1 2);
 test 4 (false 1 2);
 test 5 (cond true 1 2);
 test 6 (cond false 1 2);
 test 7 ((\ x y z -> x y z) (\ x y -> x) 1 2);
 test 8 ((\ x y z -> x y z) (\ x y -> y) 1 2)

