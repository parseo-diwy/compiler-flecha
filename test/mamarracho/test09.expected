TEST 1
	42
TEST 2
	43
TEST 3
	1
TEST 4
	2
TEST 5
	1
TEST 6
	2
TEST 7
	1
TEST 8
	2
